bundles/ch.elexis.core.ui/src/ch/elexis/core/ui/util/OrderHistoryManager.java

```diff
@@ -0,0 +1,159 @@
+
+/**
+ * Manages the order history by logging different actions such as creating,
+ * editing, ordering, and deleting orders.
+ */
+public class OrderHistoryManager {
+	private static final Logger logger = LoggerFactory.getLogger(OrderHistoryManager.class);
+
+	public void logCreateOrder(IOrder order) {
+		logOrderStatus(order, "Created", null); //$NON-NLS-1$
+	}
+
+	public void logEdit(IOrder order, IOrderEntry entry, int oldValue, int newValue) {
+		if (oldValue == newValue)
+			return;
+
+		String details = entry.getArticle().getLabel() + " changed from " + oldValue + " to " + newValue; //$NON-NLS-1$ //$NON-NLS-2$
+		logOrderStatus(order, "Edited", details); //$NON-NLS-1$
+	}
+
+	public void logDelivery(IOrder order, IOrderEntry entry, int deliveredAmount, int orderAmaunt) {
+		String details = deliveredAmount + "x von " + orderAmaunt + " " + entry.getArticle().getLabel(); //$NON-NLS-1$ //$NON-NLS-2$
+		logOrderStatus(order, "Delivered", details); //$NON-NLS-1$
+	}
+
+	public void logCreateEntry(IOrder order, IOrderEntry entry, int quantity) {
+		String details = entry.getArticle().getLabel() + "/" + quantity; //$NON-NLS-1$
+		logOrderStatus(order, "ADDMedi", details); //$NON-NLS-1$
+	}
+
+	public void logOrder(IOrder order) {
+		logOrderStatus(order, "Ordered", null); //$NON-NLS-1$
+	}
+
+	public void logDelete(IOrder order) {
+		logOrderStatus(order, "Deleted", null); //$NON-NLS-1$
+	}
+
+	public void logCompleteDelivery(IOrder order) {
+		if (order == null) {
+			return;
+		}
+
+		String details = Messages.BestellView_OrderIsClosed;
+		logOrderStatus(order, "CompleteDelivery", details); //$NON-NLS-1$
+	}
+
+	public void logRemove(IOrder order, IOrderEntry entry) {
+		if (order == null || entry == null)
+			return;
+
+		String details = entry.getArticle().getLabel() + "/" + entry.getAmount(); //$NON-NLS-1$
+		logOrderStatus(order, "RemovedMedi", details); //$NON-NLS-1$
+	}
+
+	public void logOrderSent(IOrder order, boolean sent) {
+		if (order == null)
+			return;
+
+		String method = sent ? Messages.Outputter_Sent : "Printed"; //$NON-NLS-1$
+		logOrderStatus(order, "Ordered", method); //$NON-NLS-1$
+	}
+
+	public void logSupplierAdded(IOrder order, IOrderEntry entry, String supplier) {
+		if (order == null || entry == null || supplier == null || supplier.isEmpty())
+			return;
+
+		String details = entry.getArticle().getLabel();
+		logOrderStatus(order, "SupplierAdded", details, supplier); //$NON-NLS-1$
+	}
+
+	private void logOrderStatus(IOrder order, String action, String details) {
+		logOrderStatus(order, action, details, null);
+	}
+
+	private void logOrderStatus(IOrder order, String action, String details, String extraInfo) {
+		if (order == null)
+			return;
+
+		String userId = ContextServiceHolder.get().getActiveUser().map(user -> user.getId()).orElse("Unknown"); //$NON-NLS-1$
+		OrderHistoryEntry entry = new OrderHistoryEntry(action, userId, details, extraInfo);
+		saveLogEntry(order, entry.toJson());
```

Comment:
Der entry wird hier in einen JSON String umgewandelt, und in der Methode dann wieder in ein Objekt überführt, um dann wieder in einen JSON String umgewandelt werden.

Bitte hier direkt das entry Object übergeben.